providers/base/bin/edid_cycle.py

```diff
@@ -4,152 +4,212 @@
+    def get_active_devices():
+        """Get the list of active video ouput devices."""
+        if os.getenv("XDG_SESSION_TYPE") == "wayland":
+            command = ["gnome-randr", "query"]
+            pattern = r"^\b\w+-\d+\b"
+        else:
+            command = ["xrandr", "--listactivemonitors"]
+            pattern = r"\b\w+-\d+\b$"
 
-    xrandr
-    >>> _match("Screen 0: minimum 320 x 200, current 1920 x 1080,"
-    ... " maximum 16384 x 16384i\\n"
-    ... "DPI-1 connected primary 1920x1080+0+0 (normal left inverted"
-    ... "   1920x1080     60.00*+  59.97    59.96    59.93    40.00\\n"
-    ... "HDMI-1 connected primary 1920x1080+0+0 (normal left inverted"
-    ... " right x axis y axis) 309mm x 174mm\\n"
-    ... "   1920x1080     60.00+  59.97    59.96    59.93    40.00\\n"
-    ... "   1680x1050     59.95    59.88\\n"
-    ... "   1600x1024     60.17*\\n")
-    '1600x1024'
-
-    xrandr, no output on HDMI-1
-    >>> _match("Screen 0: minimum 320 x 200, current 1920 x 1080,"
-    ... " maximum 16384 x 16384i\\n"
-    ... "DPI-1 connected primary 1920x1080+0+0 (normal left inverted"
-    ... "   1920x1080     60.00*+  59.97    59.96    59.93    40.00\\n"
-    ... "HDMI-1 connected primary 1920x1080+0+0 (normal left inverted"
-    ... " right x axis y axis) 309mm x 174mm\\n"
-    ... "   1920x1080     60.00+  59.97    59.96    59.93    40.00\\n"
-    ... "   1680x1050     59.95    59.88\\n")
-
-    xrandr, no HDMI-1 available
-    >>> _match("Screen 0: minimum 320 x 200, current 1920 x 1080,"
-    ... " maximum 16384 x 16384i\\n"
-    ... "DPI-1 connected primary 1920x1080+0+0 (normal left inverted"
-    ... "   1920x1080     60.00*+  59.97    59.96    59.93    40.00\\n")
-
-    gnome-randr
-    >>> _match("supports-mirroring: true\\n"
-    ... "layout-mode: physical\\n"
-    ... "supports-changing-layout-mode: false\\n"
-    ... "global-scale-required: false\\n"
-    ... "legacy-ui-scaling-factor: 1\\n"
-    ... 'renderer: "native"\\n\\n'
-    ... "logical monitor 0:\\n"
-    ... "x: 0, y: 0, scale: 1, rotation: normal, primary: yes\\n"
-    ... "associated physical monitors:\\n"
-    ... "	HDMI-1 DEL DELL U2722DE 4M9KFH3\\n\\n"
-    ... "HDMI-1 DEL DELL U2722DE 4M9KFH3\\n"
-    ... "              2560x1440@59.951	2560x1440 	59.95+"
-    ... "[x1.00+, x2.00, x3.00]\\n"
-    ... "              2048x1080@23.997	2048x1080 	24.00*"
-    ... "[x1.00+, x2.00]\\n")
-    '2048x1080'
-    """
-    match = re.search(PATTERN, output, re.MULTILINE | re.DOTALL)
-    if match:
-        return match.group(1)
-    return None
+        xrandr_output = subprocess.check_output(
+            command,
+            universal_newlines=True,
+            encoding="utf-8",
+            stderr=subprocess.DEVNULL,
+        )
+
+        return set(re.findall(pattern, xrandr_output, re.MULTILINE))
+
+    _set_edid(zapper_host, None)
+
+    # Not knowing what to except I can't use a busy loop here
+    time.sleep(5)
```

Comment:
because that function assumes you know the target port, I expanded the comment a bit more in https://github.com/canonical/checkbox/pull/831/commits/2678a5007d8e6f7fd6cf280873d4b24d05ba377c